src/pages/index.js

```diff
@@ -7,36 +7,85 @@ import UserInfo from "../components/UserInfo.js";
 import { initialCards } from "../utils/constants.js";
 import { config } from "../utils/constants.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
+import Popup from "../components/Popup.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+
+let section;
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "953ea6cd-2c01-4d7b-be00-e077aa224921",
+    "Content-Type": "application/json",
+  },
+});
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    setIsLiked,
+    handleDelete
+  );
   const cardElement = card.getCardView();
+
   return cardElement;
 }
 
+function setIsLiked(card) {
+  if (card.isLiked()) {
+    api.dislikeCard(card.getCardId()).then((res) => {
+      card.setIsLiked(false).catch((err) => {
+        console.error(err);
```

Comment:
Needs correction: `card.setIsLiked` doesn't return a promise. Please remove `catch` block from here and add it to the api promise

```js
api.dislikeCard(card.getCardId()).then((res) => {
  card.setIsLiked(false);
}).catch((err) => {
  // handle the error
})
```